src/numbers/min_sum_four_digit.rs

```
@@ -1,21 +1,33 @@
 pub struct Solution;
 
 impl Solution {
-    #[must_use]
-    pub fn minimum_sum(num: i32) -> i32 {
-        Self::get_sorted_digits(num)
-    }
-
-    fn get_sorted_digits(mut num: i32) -> i32 {
-        let mut digits = [0; 4];
-
-        for i in (0..4).rev() {
-            digits[i] = num % 10;
-            num /= 10;
+    /// # Errors
+    /// Returns an error if the input is not a four-digit number or if there are parsing errors.
+    pub fn minimum_sum(num: i32) -> Result<i32, &'static str> {
+        let num_str = num.to_string();
+        if num_str.len() != 4 {
+            return Err("Input must be a four-digit number");
         }
 
+        let mut digits: Vec<_> = num_str.chars().collect();
         digits.sort_unstable();
 
-        digits[0] * 10 + digits[2] + digits[1] * 10 + digits[3]
+        let (new1, new2): (String, String) =
+            digits
+                .chunks(2)
+                .fold((String::new(), String::new()), |(mut a1, mut a2), chunk| {
+                    a1.push(chunk[0]);
+                    a2.push(chunk.get(1).copied().unwrap_or('0'));
```

Instead of `unwrap_or`, it would be better to return an error if `None` because it means out of bound and something wrong.

---

BTW, `unwrap_or_default` is always preferred if it can be applied the same way.

---

Replacing `unwrap_or `with `try_fold` for better error handling during chunk processing.

---

Refer a few more comments, please.